SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebuggerEditor.cpp

```diff
@@ -0,0 +1,106 @@
+#include "Utils/SpatialDebuggerEditor.h"
+
+#include "LoadBalancing/GridBasedLBStrategy.h"
+#include "LoadBalancing/LayeredLBStrategy.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+#include "SpatialCommonTypes.h"
+#include "Utils/InspectionColors.h"
+#include "Debug/DebugDrawService.h"
+#include "Editor.h"
+#include "LevelEditor.h"
+#include "Modules/ModuleManager.h"
+#include "Engine/World.h"
+#include "GameFramework/WorldSettings.h"
+#include "EngineClasses/SpatialWorldSettings.h"
+
+using namespace SpatialGDK;
+
+ASpatialDebuggerEditor::ASpatialDebuggerEditor(const FObjectInitializer& ObjectInitializer)
+	: Super(ObjectInitializer)
+{
+	PrimaryActorTick.bCanEverTick = false;
+
+	bReplicates = false;
+}
+
+void ASpatialDebuggerEditor::Destroyed()
+{
+	DestroyWorkerRegions();
+}
+
+void ASpatialDebuggerEditor::ShowWorkerRegions(bool bEnabled)
+{
+	bShowWorkerRegions = bEnabled;
+
+	RefreshWorkerRegions();
+}
+
+
+void ASpatialDebuggerEditor::RefreshWorkerRegions()
+{
+	DestroyWorkerRegions();
+
+	if (bShowWorkerRegions && AllowWorkerBoundaries())
+	{
+		InitialiseWorkerRegions();
+		CreateWorkerRegions();
+	}
+
+	if (GEditor != nullptr && GEditor->GetActiveViewport() != nullptr)
+	{
+		// Redraw editor window to show changes
+		GEditor->GetActiveViewport()->Invalidate();
+	}
+}
+
+bool ASpatialDebuggerEditor::AllowWorkerBoundaries() const
+{
+	// Check if multi worker is enabled.
+	UWorld* World = GetWorld();
+
+	if (World == nullptr)
+	{
+		return false;
+	}
+		
+	const ASpatialWorldSettings* WorldSettings = Cast<ASpatialWorldSettings>(World->GetWorldSettings());
+
+	const bool bIsMultiWorkerEnabled = WorldSettings != nullptr && WorldSettings->IsMultiWorkerEnabled();
+
+	return bIsMultiWorkerEnabled;
+}
+
+void ASpatialDebuggerEditor::InitialiseWorkerRegions()
+{
```

Comment:
One thing we could do to remove the restriction of us only rendering the default layer AND letting users do their own visualization could be adding a method to the AbstractLBStrategy that returns a `FWorkerRegionInfo` (basically put the implementation for this method into the LayeredLbStrategy & GridBasedLBStrategy)?

Happy either way if you want to do that here or leave to a followup task?

Reply:
At the moment the editor visualisation reflects how it is currently done at run-time. I think your idea to move it to the LBStrategy is a good one but I think it would be beneficial to do it as a followup task and apply it to both editor and run-time.